File: two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hashMap = new Map
    for (let i=0; i<nums.length; i++){
        if(hashMap.has(target-nums[i])){
            return [nums.indexOf(target-nums[i]), i]
        }
        hashMap.set(nums[i], target-nums[i])
    }
};

File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        
        for (int i = 0;; i++){
            auto num = umap.find(target-nums[i]);
            
            if (num != umap.end())
                return vector<int> {i, num -> second};
            
            umap[nums[i]] = i;
        }
    }
};